src/v2/cookbook/avoiding-memory-leaks.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Avoiding Memory Leaks
+type: cookbook
+order: 9
+---
+## Introduction
+
+If you are developing applications with Vue, then you need to watch out for memory leaks. This issue is especially important in Single Page Applications (SPAs) because by design, users should not have to refresh their browser when using an SPA, so it is up to the Javascript application to clean up components and make sure that garbage collection takes place as expected.
+
+Memory leaks in Vue applications do not typically come from Vue itself, rather they can happen when incorporating other libraries into an application.
+
+## Simple Example
+
+The following example shows a memory leak caused by using the Choices.js library in a Vue component (and not properly cleaning it up). Later, we will show how to remove the Choices.js footprint and avoid the memory leak.
+
+In the example below, we load up a select with a lot of options and then we use a show/hide button with a `v-if` directive to add it and remove it from the virtual DOM. The problem with this example is that the v-if directive removes the parent element from the DOM, but we did not clean up the additional DOM pieces created by Choices.js, causing a memory leak. 
+
+To see this memory leak in action, open this [CodePen example](https://codepen.io/freeman-g/pen/qobpxo) and then, using Chrome, open the Chrome Task Manager (Chrome Menu > More Tools > Task Manager). Now, click the show/hide button 50 or so times. You should see the memory usage in the Chrome Task Manager increase and never be reclaimed.
```

Comment:
Usually we embed codepens, but I think in this case it would be really helpful to show a gif of what you're describing in action. That way they can go to the pen to see if for themselves, but they don't have to leave the flow of the recipe to see what you mean.